src/controller/OperationalCredentialsDelegate.h

```diff
@@ -81,6 +126,13 @@ class DLL_EXPORT OperationalCredentialsDelegate
      */
     virtual void SetFabricIdForNextNOCRequest(FabricId fabricId) {}
 
+    /**
+     *   This function sets the CAT values for which the next NOC Chain should be generated. This API is
+     *   not required to be implemented if the delegate implementation has other mechanisms to find the
+     *   CAT values.
+     */
+    virtual void SetCATValuesForNextNOCRequest(CATValues cats) {}
```

Comment:
This is a bad API; if we need to provide more context to NOC generation we should just provide it as part of the "sign the NOC" call, not via this out-of-band state.

Reply:
All the changes in src/controller have been reverted.